apps/src/clientApi.js

```diff
@@ -68,11 +70,180 @@ class AssetsApi extends CollectionsApi {
     });
     return ajaxInternal('POST', path, success, error);
   }
+
+  deleteFile(filename, success, error) {
+    return ajaxInternal('DELETE', this.basePath(filename), success, error);
+  }
+
+  getUploadUrl() {
+    return this.basePath('/');
+  }
+
+  makeUploadDoneCallback(callback) {
+    return callback;
+  }
+
+  makeUploadStartCallback(callback) {
+    return callback;
+  }
 }
 
+class FilesApi extends CollectionsApi {
+  getVersionHistory(success, error) {
+    var path = apiPath(
+      'files-version',
+      this.projectId || window.dashboard.project.getCurrentId()
+    );
+    return ajaxInternal('GET', path, success, error);
+  }
+
+  restorePreviousVersion(versionId, success, error) {
```

Comment:
They're fairly obvious, but it'd be nice to have doc-comments on the main API methods here.